thewalrus/_permanent.py

```diff
@@ -71,6 +71,51 @@ def perm(A, quad=True, fsum=False):
     return perm_real(A, quad=quad, fsum=fsum)
 
 
+def perm_BBFG(A):
+    """Returns the permanent of a matrix via the
+    `BBFG formula <https://en.wikipedia.org/wiki/Computing_the_permanent#Balasubramanian%E2%80%93Bax%E2%80%93Franklin%E2%80%93Glynn_formula>`_.
+
+    For more direct control, you may wish to call :func:`perm_BBFG_real`
+    or :func:`perm_BBFG_complex` directly.
+
+    Args:
+        A (array): a square array.
+
+    Returns:
+        np.float64 or np.complex128: the permanent of matrix A.
+    """
+
+    if not isinstance(A, np.ndarray):
+        raise TypeError("Input matrix must be a NumPy array.")
+
+    matshape = A.shape
+
+    if matshape[0] != matshape[1]:
+        raise ValueError("Input matrix must be square.")
+
+    if np.isnan(A).any():
+        raise ValueError("Input matrix must not contain NaNs.")
+
+    if matshape[0] == 2:
+        return A[0, 0] * A[1, 1] + A[0, 1] * A[1, 0]
+
+    if matshape[0] == 3:
+        return (
+            A[0, 2] * A[1, 1] * A[2, 0]
+            + A[0, 1] * A[1, 2] * A[2, 0]
+            + A[0, 2] * A[1, 0] * A[2, 1]
+            + A[0, 0] * A[1, 2] * A[2, 1]
+            + A[0, 1] * A[1, 0] * A[2, 2]
+            + A[0, 0] * A[1, 1] * A[2, 2]
+        )
+
+    if A.dtype == np.complex:
+        if np.any(np.iscomplex(A)):
+            return perm_BBFG_complex(A)
+        return perm_BBFG_real(np.float64(A.real))
+
+    return perm_BBFG_real(A)
+
```

Comment:
very nice addition! I would suggest that instead of writing a new function for permanents you add an optional keyword argument `method=ryser` or `method=bbfg` in the `perm` instead of creating a new function. This would be more inline with what we do for the `hafnian`.

Reply:
Thanks! Please, let me know if the new commit resolves this concern completely.